app/routes.py

```diff
@@ -86,9 +87,10 @@ def colorized_icon(
         image = Image.open(fd)
         if image.mode == 'P':
             image = image.convert('RGBA')
-        new_size = int(48 * scale)
-        if new_size != icon_set.get_default_pixel_size():
-            image = image.resize((new_size, new_size))
+
+        new_size = calculate_icon_size(image.size, scale)
+        if new_size[0] != new_size[1] or new_size[0] != icon_set.get_default_pixel_size():
```

Comment:
Why `new_size[0] != new_size[1]` ? Should it not be here `if scale != 1` ? it would make more sense in my opinion to return the icon unchanged in case of scale 1 and change it if the scale is different.

Reply:
Since the new babs icons have inconsistent sizes they will be rescaled here so that they should have the same size as the old ones

Reply:
Do I understand correctly, that currently scale = 1 means 48 px, right?
Shouldn't we keep it that way? So, if scale = 1 means 48 px.
And in case of non-quadratic icons, suggestion: smallest side is equal to 48 then while keeping the aspect ratio?

Reply:
Yes
scale = 1 means 48 px for the smaller side, unless it is the default set, where it is hardcoded that scale = 1 means 96 px

Reply:
In my opinion scale 1 should be the original size of the icon and might be different depending on the icon. We need to make sure that all new icon have a default size of 48px unless it make sense to have something different. Scaling means changing the original size of something and should not be tight to a size.

Reply:
The icons of the default set in the service are stored as 96 x 96.
If i understand correctly the icons are stored in a bigger dimension than 48 x 48 so that if the service is called with a scale bigger than 1 it will not appear pixelated since the resulting image on the viewer might be bigger than 48 x 48

Reply:
IMHO scale = 1 should always return icons of the same size in pixels.
It would be confusing, if for one icon set scale = 1 means e.g. 256x256  px and for annother icon set the same scale would yield icons of e.g. 48x48 px.
So I'd suggest we define scale = 1 means 48px for the shortes side of an icon (or instead of 48px whatever we currently use for a scale of 1). And keep that the same for all icon sets.

Also a good idea to store the raw icons in a larger size, so they can be scaled up without loosing quality.
Also: please the same way for all icon sets. Differences between the icon sets would introduca more complexity

Reply:
Basically in [https://github.com/geoadmin/doc-api-specs/pull/27/files](https://github.com/geoadmin/doc-api-specs/pull/27/files) we should specify that the size of the smaller dimension of the icon is always 48? (which should already the case in this PR)

Reply:
Let's have a chat on this on monday. I'd suggest we define the smallest side's size for all of our supported scales and store the icons only in the largest size, doing dynamic scaling on the front-end

Reply:
@hansmannj @LukasJoss currently scalling is not done in backend but by openlayer ! Viewer is always asking for scale 1x independently of the scaling asked by the user, scaling is done at frontend side. So it doesn't make any sense to keep the bigger size and do a down scalling for every icon to keep the quality it is totally non sense and add just complexity and load.